core/workspace_svg.js

```diff
@@ -19,6 +19,8 @@ goog.require('Blockly.ConnectionDB');
 /** @suppress {extraRequire} */
 goog.require('Blockly.constants');
 goog.require('Blockly.ContextMenu');
+/** @suppress {extraRequire} */
+goog.require('Blockly.ContextMenuItems');
```

Comment:
I still don't really love that this magic line is doing the initialization, and a different line somewhere else is doing it for the ShortcutRegistry (and iiuc, shouldn't there also be a similar require in block_svg? I didn't see it). The extraRequire helps, as in the tooling won't remove it, but if you're ever trying to figure out how the registry gets initialized it's confusing (and the current design also somewhat suffers from this too I think, except at least the require is not unused).

I would rather do it in requires.js so we have all of the bootstrapping-type magic requires all in the same place. Alternatively, because it seems like we definitely have other files that do this kind of thing, we could have some kind of bootstrap.js that gets called immediately and could call both `new ContextMenuRegistry()` (so that you wouldn't have to do the api change here, can still access the registry through a property instead) and `ContextMenuItems.registerDefaultItems()` and same for ShortcutRegistry. As it is this will be a breaking api change and that seems like a lot of work just to fix a dependency issue.

Sorry for being stubborn about this change, I just find this pattern really confusing, and would rather not add more of it if we can think of a better/clearer way of doing it.

Reply:
This is really just a bi-product of having self registering files, for fields since there’s no requirement to have any specific field included in the bundle we created requires.js as a place to declare what’s included in the bundle.

I’m trying to understand how registering context menu / shortcut items is different from that?

My reasoning for placing the extra require in something like blockly.js / workspace_svg is because requires in requires.js are optional you can pick and choose which you want. I think of the Blockly.js entry point as what are the things needed for Blockly to run, so I’m open to moving these to Blockly.js.

as for the idea of a bootstrap method, we have inject as one, but my question is why not keep them self-registrable? Since they are not attached to any specific workspace but global.

Reply:
I can also get behind adding comments to explain what each of the extra requires does. I don’t like that you can tell “why” it’s being required.

Reply:
> Sorry for being stubborn about this change, I just find this pattern really confusing, and would rather not add more of it if we can think of a better/clearer way of doing it.

Your concern seems warranted here.

Moving the requires into requires.js makes sense to me. Other things in there essentially do this bootstrapping, since plenty of the files end by registering themselves in the base registry.

Reply:
On second thought maybe requires.js isn’t really just for optional requires (eg: vertical / horizontal fly out), but more serves as a documented (through comments) of the various modules available to be included. Someone can always exclude something like the connection checker from their bundle, as long as they replace it with their own variation.

moving these to requires.js

Reply:
I understand how having the self-registrable bit makes it so we have to have a magic goog.require *somewhere*, I just don't like having to do it in (seemingly) random files like workspace_svg and would prefer we do it all in the same place. Having them still be self-registrable and moving the requires to either requires.js or Blockly.js would be fine with me. The distinction about requires.js being for optional features makes sense so yes maybe Blockly.js is better.

Reply:
Done.